src/Components/EnergyCalculator/Steps/ElectricityProvider.tsx

```diff
@@ -0,0 +1,148 @@
+import { zodResolver } from '@hookform/resolvers/zod';
+import { FormControl, FormHelperText, InputLabel, Select, TextField } from '@mui/material';
+import { useContext, useMemo } from 'react';
+import { Controller, SubmitHandler, useForm } from 'react-hook-form';
+import { FormattedMessage, useIntl } from 'react-intl';
+import { useParams } from 'react-router-dom';
+import { z } from 'zod';
+import useScreenApi from '../../../Assets/updateScreen';
+import { EnergyCalculatorFormData, FormData } from '../../../Types/FormData';
+import { FormattedMessageType } from '../../../Types/Questions';
+import ErrorMessageWrapper from '../../ErrorMessage/ErrorMessageWrapper';
+import PrevAndContinueButtons from '../../PrevAndContinueButtons/PrevAndContinueButtons';
+import QuestionHeader from '../../QuestionComponents/QuestionHeader';
+import { useDefaultBackNavigationFunction, useGoToNextStep } from '../../QuestionComponents/questionHooks';
+import QuestionQuestion from '../../QuestionComponents/QuestionQuestion';
+import { createMenuItems } from '../../Steps/SelectHelperFunctions/SelectHelperFunctions';
+import { Context } from '../../Wrapper/Wrapper';
+import ELECTRICITY_PROVIDERS from '../electricityProviders';
+import { useEnergyFormData } from '../hooks';
+
+export default function ElectricityProvider() {
+  const { formData, setFormData } = useContext(Context);
+  const energyDataAvailable = useEnergyFormData(formData);
+  const { uuid } = useParams();
+  const backNavigationFunction = useDefaultBackNavigationFunction('energyCalculatorElectricityProvider');
+  const nextStep = useGoToNextStep('energyCalculatorElectricityProvider');
+  const { formatMessage } = useIntl();
+  const { updateScreen } = useScreenApi();
+
+  const formSchema = z.object({
+    electricityProvider: z
+      .string({
+        errorMap: () => {
+          return {
+            message: formatMessage({
+              id: 'energyCalculator.electricityProvider.errorMessage',
+              defaultMessage: 'Please select an electricity provider',
+            }),
+          };
+        },
+      })
```

Comment:
What do you think about moving this after the min(1) and inside of a normal message object?